File: University/lib/Dto/Zystem/ZPGMDto.dart

class ZPGMDto
{
  String ZPCONO = "";
  String ZPBRNO = "";
  String ZPAPNO = "";
  String ZPPGNO = "";
  String ZPPGNA = "";
  String ZPPURL = "";
  String ZPREMA = "";
  String ZPSYST = "";
  String ZPSTAT = "";
  double ZPRCST = 0;
  double ZPCRDT = 0;
  double ZPCRTM = 0;
  String ZPCRUS = "";
  double ZPCHDT = 0;
  double ZPCHTM = 0;
  String ZPCHUS = "";

  ZPGMDto({
    this.ZPCONO = "",
    this.ZPBRNO = "",
    this.ZPAPNO = "",
    this.ZPPGNO = "",
    this.ZPPGNA = "",
    this.ZPPURL = "",
    this.ZPREMA = "",
    this.ZPSYST = "",
    this.ZPSTAT = "",
    this.ZPRCST = 0,
    this.ZPCRDT = 0,
    this.ZPCRTM = 0,
    this.ZPCRUS = "",
    this.ZPCHDT = 0,
    this.ZPCHTM = 0,
    this.ZPCHUS = "",
  });

  factory ZPGMDto.fromJson(Map<String, dynamic> json) {
    return ZPGMDto(
      ZPCONO: json["ZPCONO"] ?? "",
      ZPBRNO: json["ZPBRNO"] ?? "",
      ZPAPNO: json["ZPAPNO"] ?? "",
      ZPPGNO: json["ZPPGNO"] ?? "",
      ZPPGNA: json["ZPPGNA"] ?? "",
      ZPPURL: json["ZPPURL"] ?? "",
      ZPREMA: json["ZPREMA"] ?? "",
      ZPSYST: json["ZPSYST"] ?? "",
      ZPSTAT: json["ZPSTAT"] ?? "",
      ZPRCST: json["ZPRCST"] ?? 0,
      ZPCRDT: json["ZPCRDT"] ?? 0,
      ZPCRTM: json["ZPCRTM"] ?? 0,
      ZPCRUS: json["ZPCRUS"] ?? "",
      ZPCHDT: json["ZPCHDT"] ?? 0,
      ZPCHTM: json["ZPCHTM"] ?? 0,
      ZPCHUS: json["ZPCHUS"] ?? "",
    );
  }

  Map toMap()
  {
    var map = <String, dynamic>{};
    map["ZPCONO"] = ZPCONO;
    map["ZPBRNO"] = ZPBRNO;
    map["ZPAPNO"] = ZPAPNO;
    map["ZPPGNO"] = ZPPGNO;
    map["ZPPGNA"] = ZPPGNA;
    map["ZPPURL"] = ZPPURL;
    map["ZPREMA"] = ZPREMA;
    map["ZPSYST"] = ZPSYST;
    map["ZPSTAT"] = ZPSTAT;
    map["ZPRCST"] = ZPRCST;
    map["ZPCRDT"] = ZPCRDT;
    map["ZPCRTM"] = ZPCRTM;
    map["ZPCRUS"] = ZPCRUS;
    map["ZPCHDT"] = ZPCHDT;
    map["ZPCHTM"] = ZPCHTM;
    map["ZPCHUS"] = ZPCHUS;
    return map;
  }
}

File: University.Service/University.Dto/Zystem/ZPGMDto.cs
using System;
using System.Data;
using System.Collections.Generic;
using System.Runtime.Serialization;

using University.Dto.Base;

namespace University.Dto.Zystem
{
    [DataContract]
    public class ZPGMDto
    {
        #region General Property

        [DataMember]
        public string ZPCONO { get; set; }

        [DataMember]
        public string ZPBRNO { get; set; }

        [DataMember]
        public string ZPAPNO { get; set; }

        [DataMember]
        public string ZPPGNO { get; set; }

        [DataMember]
        public string ZPPGNA { get; set; }

        [DataMember]
        public string ZPPURL { get; set; }

        [DataMember]
        public string ZPREMA { get; set; }

        [DataMember]
        public string ZPSYST { get; set; }

        [DataMember]
        public string ZPSTAT { get; set; }

        [DataMember]
        public decimal ZPRCST { get; set; }

        [DataMember]
        public decimal ZPCRDT { get; set; }

        [DataMember]
        public decimal ZPCRTM { get; set; }

        [DataMember]
        public string ZPCRUS { get; set; }

        [DataMember]
        public decimal ZPCHDT { get; set; }

        [DataMember]
        public decimal ZPCHTM { get; set; }

        [DataMember]
        public string ZPCHUS { get; set; }

        #endregion

        #region Custom Property
    
        [DataMember]
        public int PageNumber { get; set; }

        [DataMember]
        public int PageSize { get; set; }

        [DataMember]
        public int TotalPage { get; set; }

        [DataMember]
        public int TotalRecord { get; set; }

        #endregion
    }

    public class ZPGMMappingDto : Mapper<ZPGMDto>
    {
        protected override ZPGMDto PopulateItem(IDataRecord dr)
        {
            ZPGMDto dto = new ZPGMDto();
            for (int i = 0; i < dr.FieldCount; i++)
            {
                MapProperty(dto, dr.GetName(i), dr[i]);
            }
            return dto;
        }
    }
}
